src/shipping/views/WeightRatesCreate/WeightRatesCreate.tsx

```diff
@@ -75,36 +79,56 @@ export const WeightRatesCreate: React.FC<WeightRatesCreateProps> = ({
   } = useShippingRateCreator(
     id,
     ShippingMethodTypeEnum.WEIGHT,
-    postalCodes,
-    radioInclusionType
+    state.postalCodeRules,
+    state.inclusionType
   );
 
   const handleBack = () => navigate(shippingZoneUrl(id));
 
-  const handlePostalCodeRangeAdd = (data: MinMax) => {
-    setPostalCodes(postalCodes => [
-      ...postalCodes,
-      {
-        __typename: "ShippingMethodPostalCodeRule",
-        end: data.max,
-        id: postalCodes.length.toString(),
-        inclusionType: postalCodes?.[0]?.inclusionType,
-        start: data.min
+  const onPostalCodeAssign = (rule: MinMax) => {
+    if (
+      state.postalCodeRules.filter(
+        item => item.start === rule.min && item.end === rule.max
```

Comment:
It's done 4 times, should be moved to utils function like
```suggestion
        const getPostalCodeRuleByMinMax = ({ min, max }) => ({ start, end }) => start === min && end === max
```
And then used like this
```suggestion
     state.postalCodeRules.filter(getPostalCodeRuleByMinMax(rule))
```